src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,566 @@
+"use strict";
+import d3 from "d3";
+import { parseTypedValue } from "../../../core/BaseConfig";
+import {
+    calculateVerticalPadding,
+    getXAxisXPosition,
+    isValidAxisType
+} from "../../../helpers/axis";
+import constants from "../../../helpers/constants";
+import errors from "../../../helpers/errors";
+import {
+    legendClickHandler,
+    legendHoverHandler,
+    loadLegendItem,
+    isLegendSelected
+} from "../../../helpers/legend";
+import {
+    processRegions,
+    regionLegendHoverHandler
+} from "../../../helpers/region";
+import styles from "../../../helpers/styles";
+import utils from "../../../helpers/utils";
+import { d3RemoveElement } from "../../Graph/helpers/helpers";
+import { generateColor, bubbleScale } from "./colorGradient";
+
+/**
+ * @typedef Bubble
+ */
+
+/**
+ * Transforms the point in the bubble graph on resize
+ *
+ * @private
+ * @param {object} scale - d3 scale for Graph
+ * @returns {Function} - translate function for d3 transform
+ */
+const transformPoint = (scale) => (value) => {
+    const getX = (val) => scale.x(val.x);
+    const getY = (val) => scale[val.yAxis](val.y);
+    return `translate(${getX(value)},${getY(value)})`;
+};
+/**
+ * Transforms points for a data point set in the bubble graph on resize
+ *
+ * @private
+ * @param {object} scale - d3 scale for Graph
+ * @param {d3.selection} canvasSVG - d3 selection node of canvas svg
+ * @param {string} cls - selector for the data point translation
+ * @param {object} config - config object derived from input JSON
+ * @returns {object} d3 select object
+ */
+const translatePoints = (scale, canvasSVG, cls, config) =>
+    canvasSVG
+        .selectAll(`.${styles.bubbleGraphContent} .${cls}`)
+        .each(function(d) {
+            const pointSVG = d3.select(this);
+            pointSVG
+                .transition()
+                .call(
+                    constants.d3Transition(config.settingsDictionary.transition)
+                )
+                .attr("transform", () => transformPoint(scale)(d));
+        });
+
+/**
+ * Toggles the selection of a data point, executes on click of a data point.
+ *
+ * @private
+ * @param {object} target - DOM element of the data point clicked
+ * @returns {Array} d3 html element of the selected point
+ */
+const toggleDataPointSelection = (target) => {
+    blurActionHandler(target, constants.HOVER_EVENT.MOUSE_ENTER);
```

Comment:
Hover?

Reply:
I will create a different constant for this.